IDE/STM32Cube/wolfssl_example.c

```diff
@@ -1468,25 +1469,264 @@ static void PrintMemStats(void)
 #endif
 }
 
-#if 0
-static void* wolfMallocCb(size_t size)
+/* UART DMA IO Routines */
+#ifndef B115200
+#define B115200 115200
+#endif
+
+/* Max buffer for a single TLS frame */
+#ifndef MAX_RECORD_SIZE
+#define MAX_RECORD_SIZE (16 * 1024)
+#endif
+
+typedef struct {
+	uint16_t curr_index;
+        uint16_t data_len;
+	char buf[MAX_RECORD_SIZE];
+} tls13_buf;
+
+/* This sets which UART to do the TLS 1.3 connection over.  It is something you
+ * will have to configure in STMCubeIDE and then change here. */
+#define TLS_UART huart2
+extern UART_HandleTypeDef TLS_UART;
+
+static unsigned int msg_length = 0;
+
+void HAL_UARTEx_RxEventCallback(UART_HandleTypeDef *huart, uint16_t Size) {
+   if (huart->Instance == TLS_UART.Instance) {
+       msg_length = Size;
+   }
+}
+
+static int uartIORx(WOLFSSL *ssl, char *buf, int sz, void *ctx)
 {
-	void* ptr = malloc(size);
-	if (ptr == NULL) {
-		printf("BREAK!\n");
-	}
-	return ptr;
+    HAL_StatusTypeDef status;
+    tls13_buf *tb = ctx;
+
+#ifdef DEBUG_UART_IO
+    printf("UART Read: In %d\n", sz);
+#endif
+
+    if (tb->curr_index + sz <= tb->data_len) {
```

Comment:
Please work on the signed unsigned uses here... `int` + `uint16_t` `<=`uint16_t`. Consider making the `data_len` and `curr_index` int's.

Reply:
Got it.